src/libraries/System.Private.CoreLib/src/System/Buffers/Text/Ascii.cs

```diff
@@ -0,0 +1,1244 @@
+         * operating on them.
+         */
+
+        /*
+         * Equals routines
+         *
+         * Compares two ASCII buffers for equality, optionally treating [A-Z] and [a-z] as equal.
+         * All non-ASCII bytes / chars are compared for pure binary equivalence.
+         */
+
+        /// <summary>
+        /// Returns a value stating whether the contents of two ASCII text buffers are equal.
+        /// </summary>
+        /// <param name="left">The first ASCII text buffer to compare.</param>
+        /// <param name="right">The second ASCII text buffer to compare.</param>
+        /// <returns><see langword="true"/> if the buffers are equivalent; otherwise, <see langword="false"/>.</returns>
+        /// <remarks>
+        /// The comparison is performed in a case-sensitive fashion. Calling this method is equivalent
+        /// to comparing the raw binary contents of the two buffers for equality.
+        /// </remarks>
+        public static bool Equals(ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
+            => left.SequenceEqual(right);
+
+        /// <summary>
+        /// Returns a value stating whether the contents of two ASCII text buffers are equal.
+        /// </summary>
+        /// <param name="left">The first ASCII text buffer to compare.</param>
+        /// <param name="right">The second ASCII text buffer to compare.</param>
+        /// <returns><see langword="true"/> if the buffers are equivalent; otherwise, <see langword="false"/>.</returns>
+        /// <remarks>
+        /// The comparison is performed in a case-sensitive fashion. Calling this method is equivalent
+        /// to comparing the raw binary contents of the two buffers for equality.
+        /// </remarks>
+        public static bool Equals(ReadOnlySpan<char> left, ReadOnlySpan<char> right)
+            => left.SequenceEqual(right);
+
+        /// <summary>
+        /// Returns a value stating whether the contents of two ASCII text buffers are equal
+        /// using a case-insensitive comparison.
+        /// </summary>
+        /// <param name="left">The first ASCII text buffer to compare.</param>
+        /// <param name="right">The second ASCII text buffer to compare.</param>
+        /// <returns><see langword="true"/> if the buffers are equivalent; otherwise, <see langword="false"/>.</returns>
+        public static bool EqualsIgnoreCase(ReadOnlySpan<byte> left, ReadOnlySpan<byte> right)
+        {
+            if (left.Length != right.Length)
+            {
+                return false;
+            }
+
+            for (int i = 0; i < left.Length; i++)
+            {
+                byte leftValue = left[i];
+                byte rightValue = right[i];
+
+                // The elements must be an exact match, or they must be
+                // equivalent when converted to lowercase.
+
+                if ((leftValue != rightValue) && (ToLower(leftValue) != ToLower(rightValue)))
```

Comment:
We expect the most common case to be identical casing, and so it's worth having both checks?